src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.FromJson.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization
+{
+    public static partial class JsonConverter
+    {
+        // todo: use a common instance across project (once corefxlab done)
+        internal static readonly UTF8Encoding s_utf8Encoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true);
+
+        private static readonly JsonConverterSettings s_DefaultSettings = new JsonConverterSettings();
+
+        private static bool FromJson(
+            ref Utf8JsonReader reader,
+            JsonConverterSettings settings,
+            Type returnType,
+            ref FromJsonObjectState current,
+            ref List<FromJsonObjectState> previous,
+            ref int arrayIndex)
+        {
+            while (reader.Read())
+            {
+                JsonTokenType tokenType = reader.TokenType;
+                if (tokenType == JsonTokenType.String || tokenType == JsonTokenType.Number || tokenType == JsonTokenType.True || tokenType == JsonTokenType.False)
```

Comment:
We can make this faster by doing a single range comparison between `JsonTokenType.String` and `JsonTokenType.False`

```C#
tokenType >= JsonTokenType.String && tokenType <= JsonTokenType.False
```

along with a trick like:
```C#
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static bool IsInRangeInclusive(uint value, uint lowerBound, uint upperBound)
            => (value - lowerBound) <= (upperBound - lowerBound);
```

Reply:
Sounds good although the helper method probably wouldn't be any faster, so just doing the range with a Debug.Assert for readability

Reply:
The helper method converts 2 comparisons and an and to a single comparison (i.e. fewer branches).